packages/coordinator/tests/services/verification/certificate-revocation.spec.ts

```diff
@@ -0,0 +1,353 @@
+
+      // 3.1.3 - Other handled reason code - AEA-2650/AC 1.1
+      ["Unspecified", CRLReasonCode.Unspecified],
+      ["AffiliationChanged", CRLReasonCode.AffiliationChanged],
+      ["Superseded", CRLReasonCode.Superseded],
+      ["CessationOfOperation", CRLReasonCode.CessationOfOperation],
+      ["CertificateHold", CRLReasonCode.CertificateHold],
+      ["RemoveFromCRL", CRLReasonCode.RemoveFromCRL],
+
+      // 3.1.4 - Other unhandled Reason Code - AEA-2650/comments
+      ["PrivilegeWithdrawn", CRLReasonCode.PrivilegeWithdrawn],
+      ["AACompromise", CRLReasonCode.AACompromise],
+
+      // 3.1.5 - Reason Code not specified - AEA-2650/comments
+      ["unspecified (1)", null],
+      ["unspecified (2)", ""]
+    ])("Reason Code is %s", async (desc: string, reasonCode: CRLReasonCode) => {
+      // Force an unsupported revocation value (-1) to be returned
+      const reasonCodeSpy = jest.spyOn(utils, "getRevokedCertReasonCode")
+      reasonCodeSpy.mockReturnValue(reasonCode)
+
+      const isCertificateValid = await isSignatureCertificateValid(prescription, logger)
+      expect(isCertificateValid).toEqual(false)
+
+      // Check log messages
+      expectLogMessages(reasonCode, isCertificateValid)
+
+      reasonCodeSpy.mockRestore()
+    })
+  })
+})
+
+describe("Certificate verification edge cases", () => {
+  // 4 - Unreadable certificate
+  test("unreadable certificate is not allowed", async () => {
+    const validPrescription = TestPrescriptions.parentPrescriptions.validSignature.ParentPrescription
+    const validCertificateText = common.getCertificateTextFromPrescription(validPrescription)
+    const unreadableCertText = validCertificateText.slice(1)
+
+    // make the function return the unreadable cert instead
+    const certTextSpy = jest.spyOn(common, "getCertificateTextFromPrescription")
+    certTextSpy.mockReturnValue(unreadableCertText)
+
+    const isValid = await isSignatureCertificateValid(validPrescription, logger)
+    const errorText = loggerError.mock.calls[0][0]
+    expect(isValid).toEqual(false)
+    expect(errorText).toContain("Could not parse X509 certificate from prescription")
+
+    certTextSpy.mockRestore()
+  })
+
+  // 5 - CRL Distribution Point URL not set or unavailable
+  test("CRL not set or unavailable is allowed", async () => {
+    // This cert does not have a CRL
+    const prescription = TestPrescriptions.parentPrescriptions.validSignature.ParentPrescription
+    const certificate = utils.getCertificateFromPrescription(prescription)
+    const serialNumber = utils.getX509SerialNumber(certificate)
+
+    const spy = jest.spyOn(utils, "getX509SerialNumber")
+    spy.mockReturnValue(serialNumber)
+
+    const isValid = await isSignatureCertificateValid(prescription, logger)
+    expect(isValid).toEqual(true)
+
+    const logPattern = "Cannot retrieve CRL distribution point from certificate"
+    expect(loggerError).toHaveBeenCalledWith(expect.stringContaining(logPattern))
```

Comment:
Add checks for different failures, e.g. missing CRL, network error, 404, etc.

Reply:
Add logic before!

Reply:
Added try-catch in source function, so any errors will get caught and logged with the reason.